internal/agent/metrics/metrics.go

```diff
@@ -0,0 +1,86 @@
+package metrics
+
+import (
+	"fmt"
+	"github.com/go-resty/resty/v2"
+	"github.com/npavlov/go-metrics-service/internal/storage"
+	"github.com/npavlov/go-metrics-service/internal/types"
+	"math/rand"
+	"runtime"
+)
+
+type Service interface {
+	SendMetrics()
+	UpdateMetrics()
+}
+
+type MetricService struct {
+	Storage storage.Repository
+	addr    string
+}
+
+func NewMetricService(storage storage.Repository, addr string) *MetricService {
+	return &MetricService{
+		Storage: storage,
+		addr:    addr,
+	}
+}
+
+func (m *MetricService) UpdateMetrics() {
+	var memStats runtime.MemStats
+	runtime.ReadMemStats(&memStats)
+
+	m.Storage.UpdateGauge(types.Alloc, float64(memStats.Alloc))
+	m.Storage.UpdateGauge(types.BuckHashSys, float64(memStats.BuckHashSys))
+	m.Storage.UpdateGauge(types.Frees, float64(memStats.Frees))
+	m.Storage.UpdateGauge(types.GCCPUFraction, memStats.GCCPUFraction)
+	m.Storage.UpdateGauge(types.GCSys, float64(memStats.GCSys))
+	m.Storage.UpdateGauge(types.HeapAlloc, float64(memStats.HeapAlloc))
```

Comment:
[CLARIFICATION] Please read the task again. Why did you convert everything to `float64`?

Reply:
A new structure Metrics has been created for such purposes